pl_bolts/losses/object_detection.py

```diff
@@ -0,0 +1,40 @@
+"""
+Generalized Intersection over Union (GIoU) loss (Rezatofighi et. al)
+"""
+
+import torch
+
+
+def giou_loss(pred: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
+    """
+    Calculates the generalized intersection over union loss.
+
+    It has been proposed in `Generalized Intersection over Union: A Metric and A
+    Loss for Bounding Box Regression <https://arxiv.org/abs/1902.09630>`_.
+
+    Args:
+        pred: batch of prediction bounding boxes with representation
+              [x_min, y_min, x_max, y_max]
+        target: batch of target bounding boxes with representation
+                [x_min, y_min, x_max, y_max]
+
+    Returns:
+        GIoU loss
+    """
```

Comment:
pls add doctest example

Reply:
Forgive my ignorance, but how do I do that?

Reply:
I think @Borda means something like:
https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/58536c260e7877c8c75dfccfaa58acee4fdc2ae1/pl_bolts/datasets/cifar10_dataset.py#L36-L50